graphql-dgs-spring-boot-oss-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsAutoConfiguration.kt

```diff
@@ -307,14 +305,8 @@ open class DgsAutoConfiguration(
         havingValue = "false",
         matchIfMissing = false
     )
-    open fun noIntrospectionFieldVisibility(): GraphqlFieldVisibility {
-        return NO_INTROSPECTION_FIELD_VISIBILITY
-    }
-
-    @Bean
-    @ConditionalOnMissingBean
-    open fun defaultFieldVisibility(): GraphqlFieldVisibility {
-        return DEFAULT_FIELD_VISIBILITY
+    open fun noIntrospectionFieldVisibility() {
+        Introspection.enabledJvmWide(false)
```

Comment:
I know that `NoIntrospectionGraphqlFieldVisibility` is deprecated, and suggests using this static / global setting. But affecting the state of the entire JVM as a side effect of a bean method seems a little off to me.

There's also the `Introspection.INTROSPECTION_DISABLED` String which can be put in the `GraphQlContext` to control this at a per-request level. Have we considered instead registering a `GraphQLContextContributor` that sets this context instead?

Reply:
Re: affecting the state of the entire JVM as a side effect of a bean, that bean is for configuring introspection and, to my understanding, there one executable schema which queries are executed against in the dgs application. However, I do see that affecting the entire JVM could be too restrictive for more advanced usecases where there might be multiple executable schema running in the application or other cases (like the per request) where introspection is set dynamically. 

I have not looked into `Introspection.INTROSPECTION_DISABLED`, I'll take a look, thanks! In this case though the purpose of using Instrospection.disabledJvmWide was to be able to disable introspection via the spring property without adding in a deprecated method (for the case where users use that flag to disable introspection). I'm thinking the approach to take here might be the alternative approach considered, which is throwing an exception to prefer the dgs config as disabling at a per request level might be a bigger change/feature add for this case

Reply:
Doing what I suggested would not use any deprecated classes / methods.

What I'm suggesting is doing something like this:

```kotlin
    @Bean
    @ConditionalOnProperty(
        prefix = "$AUTO_CONF_PREFIX.introspection",
        name = ["enabled"],
        havingValue = "false",
        matchIfMissing = false
    )
    open fun disableIntrospectionContextContributor(): GraphQLContextContributor {
        return GraphQLContextContributor { builder, _, _ ->
            builder.put(Introspection.INTROSPECTION_DISABLED, true)
        }
    }
```

Reply:
I think this approach from @kilink is quite nice. The global JVM setting is a bit weird.

Reply:
+1 to Patrick's approach.

Reply:
+1 very nice approach. Tried it out and updated it for both spring and dgs introspection enabled properties.